kubernetes-deployment/maintainlib/etcdfix.py

```diff
@@ -0,0 +1,236 @@
+
+        commandline = "tar -xvf {0}.tar && sudo ./{0}/stop-etcd-server.sh".format("etcd-reconfiguration-stop")
+
+        if common.ssh_shell_paramiko(bad_node_config, commandline) == False:
+            return
+
+        self.logger.info("Successfully stoping bad etcd server on node {0}".format(bad_node_config["nodename"]))
+
+        if self.clean_flag:
+            self.delete_packege(bad_node_config)
+
+
+
+    def update_etcd_cluster(self, good_node_config, bad_node_config):
+
+        self.prepare_package(good_node_config, "etcd-reconfiguration-update")
+
+        self.logger.info("Begin to execute the job : etcd-reconfiguration-update.")
+        self.logger.info("Update etcd cluster on host [{0}].".format(good_node_config['nodename']))
+
+        script_package = "etcd-reconfiguration-update.tar"
+        src_local = "parcel-center/{0}".format(good_node_config["nodename"])
+        dst_remote = "/home/{0}".format(good_node_config["username"])
+
+        if common.sftp_paramiko(src_local, dst_remote, script_package, good_node_config) == False:
+            return
+
+        commandline = "tar -xvf {0}.tar && sudo ./{0}/{1}.sh {2} {3}".format("etcd-reconfiguration-update", "update-etcd-cluster", bad_node_config['hostip'], bad_node_config['etcdid'] )
+
+        if common.ssh_shell_paramiko(good_node_config, commandline) == False:
+            return
+
+        self.logger.info("Successfully update etcd cluster configuration on node {0}".format(bad_node_config["nodename"]))
+
+        if self.clean_flag:
+            self.delete_packege(good_node_config)
+
+
+
+    def restart_etcd_server(self, bad_node_config):
+
+        self.logger.info("Begin to execute the job : etcd-reconfiguration-restart.")
+        self.logger.info("Restart etcd server on host [{0}].".format(bad_node_config['nodename']))
+
+        new_etcd_cluster_ips_peer = self.get_etcd_peer_ip_list(bad_node_config)
+
+        self.cluster_config['clusterinfo']['etcd_cluster_ips_peer'] = new_etcd_cluster_ips_peer
+        self.cluster_config['clusterinfo']['etcd-initial-cluster-state'] = 'existing'
+
+        self.prepare_package(bad_node_config, "etcd-reconfiguration-restart")
+
+        script_package = "etcd-reconfiguration-restart.tar"
+        src_local = "parcel-center/{0}".format(bad_node_config["nodename"])
+        dst_remote = "/home/{0}".format(bad_node_config["username"])
+
+        if common.sftp_paramiko(src_local, dst_remote, script_package, bad_node_config) == False:
+            return
+
+        commandline = "tar -xvf {0}.tar && sudo ./{0}/{1}.sh".format("etcd-reconfiguration-restart", "restart-etcd-server")
+
+        if common.ssh_shell_paramiko(bad_node_config, commandline) == False:
+            return
+
+        self.logger.info("Successfully restarting bad etcd server on node {0}".format(bad_node_config["nodename"]))
+
+        if self.clean_flag:
+            self.delete_packege(bad_node_config)
+
+
+
+    def get_etcd_leader_node(self):
```

Comment:
>get_etcd_leader_node [](start = 8, length = 20)

I suggest you move etcd related operation to common lib

Reply:
Better not. I hope one lib's modify won't have an influence on others.